File: kata-solutions/tell-dont-ask/tell-dont-ask-java/app/src/main/java/domain/OrderItem.java
package domain;

import java.math.BigDecimal;
import static java.math.RoundingMode.HALF_UP;

public class OrderItem {
    public final Product product;
    public final int quantity;
    public final BigDecimal taxedAmount;
    public final BigDecimal tax;

    public OrderItem(Product product, int quantity) {
        this.product = product;
        this.quantity = quantity;
        this.taxedAmount = product.unitaryTaxedAmount().multiply(BigDecimal.valueOf(quantity)).setScale(2, HALF_UP);
        this.tax = product.unitaryTax().multiply(BigDecimal.valueOf(quantity));
    }
}

File: tdd-katas/tell-dont-ask/tell-dont-ask-typescript/src/domain/OrderItem.ts
import Product from "./Product";
import bigDecimal from "js-big-decimal";

export default class OrderItem {
    private _product: Product = new Product();
    private _quantity: number = 0;
    private _taxedAmount: bigDecimal = new bigDecimal(0);
    private _tax: bigDecimal = new bigDecimal(0);

    constructor() {}

    get product(): Product {
        return this._product;
    }

    set product(value: Product) {
        this._product = value;
    }

    get quantity(): number {
        return this._quantity;
    }

    set quantity(value: number) {
        this._quantity = value;
    }

    get taxedAmount(): bigDecimal {
        return this._taxedAmount;
    }

    set taxedAmount(value: bigDecimal) {
        this._taxedAmount = value;
    }

    get tax(): bigDecimal {
        return this._tax;
    }

    set tax(value: bigDecimal) {
        this._tax = value;
    }
}